packages/progress-circle/src/ProgressCircle.ts

```diff
@@ -18,17 +18,24 @@ import {
     SpectrumElement,
     TemplateResult,
 } from '@spectrum-web-components/base';
-import { property } from '@spectrum-web-components/base/src/decorators.js';
+import {
+    property,
+    query,
+} from '@spectrum-web-components/base/src/decorators.js';
+import { ObserveSlotText } from '@spectrum-web-components/shared/src/observe-slot-text.js';
 import { ifDefined } from '@spectrum-web-components/base/src/directives.js';
 
 import progressCircleStyles from './progress-circle.css.js';
 
 /**
  * @element sp-progress-circle
  */
-export class ProgressCircle extends SizedMixin(SpectrumElement, {
-    validSizes: ['s', 'm', 'l'],
-}) {
+export class ProgressCircle extends SizedMixin(
+    ObserveSlotText(SpectrumElement, ''),
```

Comment:
In that Progress Circle doesn't show the label visibly, it might not need this feature addition. It's [not "free" from a performance standpoint](https://github.com/adobe/spectrum-web-components/pull/3483#user-content-progress-circle), so we should be sure that it is 100% needed.

Progress Bar _may_ need it, as the label is visible there.

Reply:
what could be the other way of allowing the innerText to act as a label in case label is not provided without using a slot?

Reply:
`ObserveSlotText` will only let you know if there is text there or not in a reactive way. You do much the same in `getLabelFromSlot` right now.